mediatek: filogic: add support for COMFAST CF-WR632AX

koreapyj opened the pull request:
Add support for COMFAST CF-WR632AX

Hardware:
- SoC: Mediatek MT7981
- RAM: DDR3 512MB
- Flash: WINBOND W25N01GV (SPI-NAND 128MB)
- WiFi: Mediatek MT7976CN
- Ether0: WAN GPY211 (10/100/1000/2.5G)
- Ether1: LAN SoC (10/100/1000)
- UART: RX TX GND VCC (115200)

MAC addr:
- LAN : Label MAC
- WAN : Label MAC + 1

Installation:
1. Unplug router and plug ethernet cable on "LAN" port.
2. Hold reset button and plug in USB-C power cable.
3. Wait until green LED flashes 5 times and release reset button. Please note that if you wait more than 7 times you will get U-Boot Netconsole instead of Web UI.
4. Open your browser and browse to http://192.168.1.1
5. Upload **squashfs-sysupgrade.bin** and done.

Rondom commented on target/linux/mediatek/filogic/base-files/etc/hotplug.d/ieee80211/11_fix_wifi_mac:
```diff
@@ -76,6 +76,7 @@ case "$board" in
 		addr=$(cat /sys/class/net/eth0/address)
 		[ "$PHYNBR" = "1" ] && macaddr_add $addr -1 > /sys${DEVPATH}/macaddress
 		;;
+	comfast,cf-wr632ax|\
```
Can this use NVMEM instead?

koreapyj commented:
> Can you also share bin files? and you flashed device with fail safe mode?

https://media.dcmys.net/raw/openwrt-mediatek-filogic-comfast_cf-wr632ax-initramfs-kernel.bin

I tested with UART U-Boot and sysupgrade. If you wish to test with stock upgrade way I can provide sample factory image. Currently I can't test it since I don't have any stock firmware image and vendor did not provided any firmware images.

koreapyj commented:
> > I tested with UART U-Boot and sysupgrade.
> 
> Can you, please, tell more details?Thank you!

Forget about previous reply. This device uses standard openwrt sysupgrade-tar image. Please refer installation instruction on https://github.com/openwrt/openwrt/pull/20244#issue-3467759863 and use this image.

https://media.dcmys.net/raw/openwrt-24.10.3-mediatek-filogic-comfast_cf-wr632ax-squashfs-sysupgrade.bin

charlesgek commented:
@koreapyj  I also had the same issue with your image, no reaction after update, fail to give IP, and when forced, just nothing. feels like bricked. had to force u-boot to flash another image.

koreapyj committed:
mediatek: filogic: add support for COMFAST CF-WR632AX

Add support for COMFAST CF-WR632AX

Hardware:
- SoC: Mediatek MT7981
- RAM: DDR3 512MB
- Flash: WINBOND W25N01GV (SPI-NAND 128MB)
- WiFi: Mediatek MT7976CN
- Ether0: WAN GPY211 (10/100/1000/2.5G)
- Ether1: LAN SoC (10/100/1000)
- UART: RX TX GND VCC (115200)

MAC addr:
- LAN : Label MAC
- WAN : Label MAC + 1

Installation:
1. Unplug router and plug ethernet cable on "LAN" port.
2. Hold reset button and plug in USB-C power cable.
3. Wait until green LED flashes 5 times and release reset button. Please note that if you wait more than 7 times you will get U-Boot Netconsole instead of Web UI.
4. Open your browser and browse to http://192.168.1.1
5. Upload **squashfs-sysupgrade.bin** and done.

Signed-off-by: Yoonji Park <koreapyj@dcmys.kr>

hauke commented on target/linux/mediatek/image/filogic.mk:
```diff
@@ -753,6 +753,29 @@ define Device/comfast_cf-e393ax
+  DEVICE_DTS := mt7981a-comfast-cf-wr632ax
+  DEVICE_DTS_DIR := ../dts
+  DEVICE_DTC_FLAGS := --pad 4096
+  DEVICE_DTS_LOADADDR := 0x43f00000
```
Is this needed?

hauke commented on target/linux/mediatek/image/filogic.mk:
```diff
@@ -753,6 +753,29 @@ define Device/comfast_cf-e393ax
+define Device/comfast_cf-wr632ax
+  DEVICE_VENDOR := COMFAST
+  DEVICE_MODEL := CF-WR632AX
+  DEVICE_DTS := mt7981a-comfast-cf-wr632ax
```
Is this really a `mt7981a` chip and not `mt7981b`?
